client/src/Pages/Dashboard/DashboardView.js

```diff
@@ -1,72 +1,217 @@
 import React, { Component } from "react";
-import GoogleMap from "../../Components/GoogleMap/GoogleMap";
-import AntdTableComponent from "../../Components/AntdTable/AntdTableComponent";
+import { graphql } from "react-apollo";
+import { getDashboardAllDataQuery } from "../../queris/queris";
+
 import { Card } from 'semantic-ui-react';
-// import CarList from "../../Components/CarList"
-import CarNameCard from "../../Components/Card/CarNameCard";
+import LoadingSpinner from '../../Components/LoadingSpinner';
 import LastFillupCard from "../../Components/Card/LastFillupCard";
-// import LastFillupTimeCard from "../../Components/Card/LastFillupTimeCard";
 import FuelLeftCard from "../../Components/Card/FuelLeftCard";
-// import TraveldSinceCard from "../../Components/Card/TraveldSinceCard";
 import DiagnosticCard from "../../Components/Card/DiagnosticCard";
-// import DiagnosticDetailCard from "../../Components/Card/DiagnosticDetailCard";
 import BusinessRatioCard from "../../Components/Card/BusinessRatioCard";
-// import BusinessTotalCard from "../../Components/Card/BusinessTotalCard";
 import AverageSpeedCard from "../../Components/Card/AverageSpeedCard";
 import TravelDistanceTotalCard from "../../Components/Card/TravelDistanceTotalCard";
-// import TravelDistanceThisYearCard from "../../Components/Card/TravelDistanceThisYearCard";
 import TimeInCarCard from "../../Components/Card/TimeInCarCard";
 import EmissionsCard from "../../Components/Card/EmissionsCard";
 import FuelEconomyCard from "../../Components/Card/FuelEconomyCard";
-// import ParkingCard from "../../Components/Card/ParkingCard";
-// import TimeTraveldCard from "../../Components/Card/TimeTraveldCard";
-// import StartLocationCard from "../../Components/Card/StartLocationCard";
-// import EndLocationCard from "../../Components/Card/EndLocationCard";
-import SmartCard from "../../Components/Card/SmartCard";
+
+import GoogleMap from "../../Components/GoogleMap/GoogleMap";
+import AntdTableComponent from "../../Components/AntdTable/AntdTableComponent";
 
 
 
 class DashboardView extends Component {
+
+
+
+  displayLastFillupData() {
+    let data = this.props.data;
+    if (data.loading) {
+      return <LoadingSpinner />;
```

Comment:
I wouldn't do this. I'd have one loading spinner for the entire dashboard.

```
render() {
    if (this.props.data.loading) {
      return <LoadingSpinner />;
    }
    return ...
```

Reply:
Make it one loading spinner.
Having a problem with the size of loading spinner which overlaps DashboardSidebar